### DISCUSSION TITLE
how to check if setting is a sub json before update default json file

### DISCUSSION THREAD

**Discussion by xiayh1992** [nlohmann/json#3613] [Q&A] [answered]
the update API can update local default json file in general.  but sometime the local json file is self-contained, it mean the local default json file not need to be added any new key ,  it just need to refresh the value .
so before refresh json file , i need check if the user setting json file is a sub of local default json file. 
is there any api to check if a json is a sub of another.
for example,
local_default_json = {"param1": 2, "param2": 3};
user_setting_json = {"param1": 1};
then i can update local_default_json by local_default_json.update(user_setting_json ) to get local_default_json = {"param1": 1, "param2": 3};

but if user_setting_json = {"param1": 1, "param3": 3}; 
then i need a function to tell user_setting_json is not a legal setting_json, because it have a new key "param3".
thanks!

**Comment by falbrechtskirchinger** [comment, marked as answer]
You'll have to manually check that your JSON object only contains valid keys.
For example, by iterating your object using https://json.nlohmann.me/api/basic_json/items/.

**Reply by xiayh1992** [reply]
my user_setting_json is a nested json file ,.
is a good idea to use user_setting_json.flatten() to get all key and then use the item() api. otherwise I need check if current json is primitive or object, to get the outlier key. what's your advise.
thank your quick answer.

**Reply by falbrechtskirchinger** [reply]
`flatten()` would work. Here's a quick demo:

https://godbolt.org/z/T6rxzbKKn

**Reply by nlohmann** [reply]
@xiayh1992 Do you need further assistance?

**Reply by xiayh1992** [reply]
thanks ,I have solved it , I appreciate you and your project .
